x/wasm/keeper/keeper_test.go

```diff
@@ -2489,6 +2489,58 @@ func TestGasConsumed(t *testing.T) {
 	}
 }
 
+func TestSetContractLabel(t *testing.T) {
+	parentCtx, keepers := CreateTestInput(t, false, AvailableCapabilities)
+	k := keepers.WasmKeeper
+	example := InstantiateReflectExampleContract(t, parentCtx, keepers)
+
+	specs := map[string]struct {
+		newLabel string
+		caller   sdk.AccAddress
+		policy   types.AuthorizationPolicy
+		expErr   bool
+	}{
+		"update label - default policy": {
+			newLabel: "new label",
+			caller:   example.CreatorAddr,
+			policy:   DefaultAuthorizationPolicy{},
+		},
+		"update label - unauthorized": {
+			newLabel: "new label",
+			caller:   RandomAccountAddress(t),
+			policy:   DefaultAuthorizationPolicy{},
+			expErr:   true,
+		},
+		"update label - gov policy": {
+			newLabel: "new label",
+			policy:   GovAuthorizationPolicy{},
+			caller:   example.CreatorAddr,
```

Comment:
When you use a random caller address, then it would ensure the different behaviour of the gov policy